components/Portfolio/Trading/TradeHistory/index.tsx

```diff
@@ -1,96 +1,63 @@
-import React from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import { Logo } from '@components/General';
-import { toApproxCurrency } from '@libs/utils';
-import { TableHeading, TableRow, TableCell } from '@components/Portfolio';
-import { DateAndTime } from '@components/General';
+import { TableHeading, TableRow, TableCell, Table } from '@components/Table';
+import { timeAgo, toApproxCurrency } from '@libs/utils';
+import { LabelledOrders } from '@libs/types/OrderTypes';
 
-const TradeHistory: React.FC = () => {
-    const headings = ['Date', 'Market', 'Position', 'Exposure / Price', 'Slippage', 'Fees', 'Total Cost', 'Order Type'];
+interface THProps {
+    fetchedTracers: any;
+    allFilledOrders: LabelledOrders;
+}
+const TradeHistory: FC<THProps> = ({ fetchedTracers, allFilledOrders }: THProps) => {
+    const [orderHistory, setOrderHistory] = useState<any>([]);
 
-    const tracers = [
-        {
-            date: '24/04/2021',
-            time: '04:31pm',
-            name: 'TSLA',
-            market: 'TSLA-USDC',
-            position: 'long',
-            exposure: 4.5,
-            slippage: 3.23,
-            fees: 2.23,
-            cost: 453.23,
-            type: 'Market',
-        },
-        {
-            date: '24/04/2021',
-            time: '04:31pm',
-            name: 'TSLA',
-            market: 'TSLA-USDC',
-            position: 'long',
-            exposure: 4.5,
-            slippage: 3.23,
-            fees: 2.23,
-            cost: 453.23,
-            type: 'Market',
-        },
-        {
-            date: '24/04/2021',
-            time: '04:31pm',
-            name: 'TSLA',
-            market: 'TSLA-USDC',
-            position: 'long',
-            exposure: 4.5,
-            slippage: 3.23,
-            fees: 2.23,
-            cost: 453.23,
-            type: 'Market',
-        },
-    ];
-
-    const TableHeadEndTheme = {
-        minWidth: '200px',
-        borderBottom: '1px solid var(--color-accent)',
-    };
+    useEffect(() => {
+        const tempOrders: any[] = [];
+        fetchedTracers.map((tracer: any) => {
+            if (allFilledOrders[tracer.address] !== []) {
+                tempOrders.push(allFilledOrders[tracer.address]);
+            }
+        });
+        tempOrders.sort((order1, order2) => (order1.timestamp < order2.timestamp && 1) || -1);
+        setOrderHistory(tempOrders);
+    }, [fetchedTracers]);
```

Comment:
Do you need to filter here? isnt it already filtered from the graph?

Reply:
yeah you're right, probably don't need to filter here